src/cephadm/cephadm

```diff
@@ -4563,9 +4563,19 @@ def list_daemons(ctx, detail=True, legacy_dir=None):
                                                      'alertmanager',
                                                      'node-exporter']:
                                     cmd = daemon_type.replace('-', '_')
-                                    out, err, code = call(ctx,
-                                        [container_path, 'exec', container_id,
-                                         cmd, '--version'])
+                                    if daemon_type == 'alertmanager':
+                                        for cmd in ['alertmanager', 'prometheus-alertmanager']:
+                                            out, err, code = call(ctx, [
+                                                    container_path, 'exec', container_id, cmd,
+                                                    '--version'
+                                                ], verbosity=CallVerbosity.SILENT)
+                                            if code == 0:
+                                                break
+                                        cmd = 'alertmanager'  # reset cmd for version extraction
+                                    else:
+                                        out, err, code = call(ctx, [
+                                            container_path, 'exec', container_id, cmd, '--version'
+                                        ])
```

Comment:
this function is already becoming incredibly hard to read ... 

any chance we can move the version logic up into a staticmethod of the `Monitoring` class?

https://github.com/ceph/ceph/blob/80fb7538756820c5d996734e21935ae6fdff14cb/src/cephadm/cephadm#L362-L373